Tests/WorkspaceTests/PrebuiltsTests.swift

```diff
@@ -0,0 +1,545 @@
+            )
+        )
+
+        try await workspace.checkPackageGraph(roots: ["Foo"]) { modulesGraph, diagnostics in
+            XCTAssertTrue(diagnostics.filter({ $0.severity == .error }).isEmpty)
+            let rootPackage = try XCTUnwrap(modulesGraph.rootPackages.first)
+            let macroTarget = try XCTUnwrap(rootPackage.underlying.modules.first(where: { $0.name == "FooMacros" }))
+            try checkSettings(macroTarget, usePrebuilt: false)
+            let testTarget = try XCTUnwrap(rootPackage.underlying.modules.first(where: { $0.name == "FooTests" }))
+            try checkSettings(testTarget, usePrebuilt: false)
+        }
+    }
+
+    func testUnsupportedArch() async throws {
+        let sandbox = AbsolutePath("/tmp/ws")
+        let fs = InMemoryFileSystem()
+
+        let artifact = Data()
+        let (manifest, rootPackage, swiftSyntax) = try initData(artifact: artifact, swiftSyntaxVersion: "600.0.1")
+        let manifestData = try JSONEncoder().encode(manifest)
+
+        let httpClient = HTTPClient { request, progressHandler in
+            guard case .download(let fileSystem, let destination) = request.kind else {
+                throw StringError("invalid request \(request.kind)")
+            }
+
+            if request.url == "https://github.com/dschaefer2/swift-syntax/releases/download/600.0.1/\(self.swiftVersion)-manifest.json" {
+                try fileSystem.writeFileContents(destination, data: manifestData)
+                return .okay()
+             } else {
+                XCTFail("Unexpected URL \(request.url)")
+                return .notFound()
+            }
+        }
+
+        let archiver = MockArchiver(handler: { _, archivePath, destination, completion in
+            XCTFail("Unexpected call to archiver")
+            completion(.success(()))
+        })
+
+        let workspace = try await MockWorkspace(
+            sandbox: sandbox,
+            fileSystem: fs,
+            roots: [
+                rootPackage
+            ],
+            packages: [
+                swiftSyntax
+            ],
+            prebuiltsManager: .init(
+                httpClient: httpClient,
+                archiver: archiver
+            ),
+            customHostTriple: Triple("86_64-unknown-linux-gnu")
+        )
+
+        try await workspace.checkPackageGraph(roots: ["Foo"]) { modulesGraph, diagnostics in
+            XCTAssertTrue(diagnostics.filter({ $0.severity == .error }).isEmpty)
+            let rootPackage = try XCTUnwrap(modulesGraph.rootPackages.first)
+            let macroTarget = try XCTUnwrap(rootPackage.underlying.modules.first(where: { $0.name == "FooMacros" }))
+            try checkSettings(macroTarget, usePrebuilt: false)
+            let testTarget = try XCTUnwrap(rootPackage.underlying.modules.first(where: { $0.name == "FooTests" }))
+            try checkSettings(testTarget, usePrebuilt: false)
+        }
+    }
+
+    func testUnsupportedSwiftVersion() async throws {
```

Comment:
Question: I'm on mobile at the moment, but how is this test different than the "testUnsupportedSwiftSyntaxVersion" ?

Reply:
This is for the toolchain version.

Reply:
Here is the diff between `testUnsupportedSwiftVersion()` and `testUnsupportedSwiftSyntaxVersion`.  Only the Swift Syntax version number appears to be different.   Considering the `let httpClient` is the same, except for the version number in the URL, is the test essentially the same?

```
diff --git a/foo.swift b/foo.swift
index 9bd41be72..10719a069 100644
--- a/foo.swift
+++ b/foo.swift
@@ -1,13 +1,12 @@
-    func testUnsupportedSwiftVersion() async throws {
+    func testUnsupportedSwiftSyntaxVersion() async throws {
          let sandbox = AbsolutePath("/tmp/ws")
          let fs = InMemoryFileSystem()
 
          let artifact = Data()
-         let (_, rootPackage, swiftSyntax) = try initData(artifact: artifact, swiftSyntaxVersion: "600.0.1")
+         let (_, rootPackage, swiftSyntax) = try initData(artifact: artifact, swiftSyntaxVersion: "600.0.2")
 
          let httpClient = HTTPClient { request, progressHandler in
-             if request.url == "https://github.com/dschaefer2/swift-syntax/releases/download/600.0.1/\(self.swiftVersion)-manifest.json" {
-                 // Pretend it's a different swift version
+             if request.url == "https://github.com/dschaefer2/swift-syntax/releases/download/600.0.2/\(self.swiftVersion)-manifest.json" {
                  return .notFound()
               } else {
                  XCTFail("Unexpected URL \(request.url)")
@@ -43,5 +42,4 @@
              let testTarget = try XCTUnwrap(rootPackage.underlying.modules.first(where: { $0.name == "FooTests" }))
              try checkSettings(testTarget, usePrebuilt: false)
          }
-
      }
\ No newline at end of file
```

Reply:
Ah, I didn't get around to mocking the swiftVersion. I'll take a look.

Reply:
No, wait, the comment in the http client is important. It's pretending we don't have the manifest for the swiftVersion being requested.